src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/WriteHelper.cs

```diff
@@ -735,6 +735,67 @@ public async Task<int> CopyKeyValueStoreSchemaHistory(ICollection<KeyValueStoreS
         return entities.Count;
     }
 
+    public async Task<int> CopyResourceOwners(ICollection<ResourceOwners> entities, CancellationToken token)
+    {
+        if (!entities.Any())
+        {
+            return 0;
+        }
+
+        var sw = Stopwatch.GetTimestamp();
+
+        await using var createTempTableCommand = _connection.CreateCommand();
+        createTempTableCommand.CommandText = @"
+CREATE TEMP TABLE tmp_resource_owners
+(LIKE resource_owners INCLUDING DEFAULTS)
+ON COMMIT DROP";
+
+        await createTempTableCommand.ExecuteNonQueryAsync(token);
+
+        await using var writer =
+            await _connection.BeginBinaryImportAsync(
+                "COPY tmp_resource_owners (id, entity_id, resource_entity_id, discriminator, balance, total_count) FROM STDIN (FORMAT BINARY)",
+                token);
+
+        foreach (var e in entities)
+        {
+            await writer.StartRowAsync(token);
+            await writer.WriteAsync(e.Id, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.EntityId, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.ResourceEntityId, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(GetDiscriminator<ResourceType>(e.GetType()), "resource_type", token);
+
+            if (e is FungibleResourceOwners fe)
+            {
+                await writer.WriteAsync(fe.Balance.GetSubUnitsSafeForPostgres(), NpgsqlDbType.Numeric, token);
+                await writer.WriteNullAsync(token);
+            }
+            else if (e is NonFungibleResourceOwners nfe)
+            {
+                await writer.WriteNullAsync(token);
+                await writer.WriteAsync(nfe.TotalCount, NpgsqlDbType.Bigint, token);
+            }
+        }
+
+        await writer.CompleteAsync(token);
+        await writer.DisposeAsync();
+
+        await using var copyFromTempTablecommand = _connection.CreateCommand();
+        copyFromTempTablecommand.CommandText = @"
```

Comment:
I hoped we'll not execute this as part of the regular DA flow as it may be really slow. How about:
1. Create long-lived temp table for those writes. i.e. just a regular table that the DA will append data to.
2. Move this logic to some db-level trigger.

This way DA won't need to wait for this operation to complete.

The query would remain essentially identical, maybe with an extra `WHERE old.from_state_version < new.from_state_version` check

Reply:
After discussion we decided to go with current solution and check performance on some real network.